### PR TITLE
Scaleway init and nodeup

### PR DESCRIPTION
**Pull request by Mia-Cross**
The goal of this PR is to initialize Scaleway support in kOps :

- by allowing the feature flag "Scaleway"
- by adding the right environment variables to the bootstrap script (therefore they will also be passed to nodeup)

### CODE REVIEW

File: pkg/model/bootstrapscript.go [hunk 1]
```diff
@@ -213,6 +214,34 @@ func (b *BootstrapScript) buildEnvironmentVariables(cluster *kops.Cluster) (map[
 		}
 	}
 
+	if cluster.Spec.GetCloudProvider() == kops.CloudProviderScaleway {
+		region, err := scw.ParseRegion(os.Getenv("SCW_DEFAULT_REGION"))
+		if err != nil {
+			return nil, fmt.Errorf("error parsing SCW_DEFAULT_REGION: %w", err)
+		}
+		env["SCW_DEFAULT_REGION"] = string(region)
+		zone, err := scw.ParseZone(os.Getenv("SCW_DEFAULT_ZONE"))
```
Comment (remyleone) [inline pkg/model/bootstrapscript.go:223]:
```suggestion
		env["SCW_DEFAULT_REGION"] = string(region)
		
		zone, err := scw.ParseZone(os.Getenv("SCW_DEFAULT_ZONE"))
```

File: pkg/model/bootstrapscript.go [hunk 2]
```diff
@@ -213,6 +214,34 @@ func (b *BootstrapScript) buildEnvironmentVariables(cluster *kops.Cluster) (map[
...
+		if err != nil {
+			return nil, fmt.Errorf("error parsing SCW_DEFAULT_ZONE: %w", err)
+		}
+		env["SCW_DEFAULT_ZONE"] = string(zone)
+		scwAccessKey := os.Getenv("SCW_ACCESS_KEY")
```
Comment (remyleone) [inline pkg/model/bootstrapscript.go:228]:
```suggestion
		env["SCW_DEFAULT_ZONE"] = string(zone)
		
		scwAccessKey := os.Getenv("SCW_ACCESS_KEY")
```

File: pkg/model/bootstrapscript.go [hunk 3]
```diff
@@ -213,6 +214,34 @@ func (b *BootstrapScript) buildEnvironmentVariables(cluster *kops.Cluster) (map[
...
+		if scwAccessKey == "" {
+			return nil, fmt.Errorf("SCW_ACCESS_KEY has to be set as an environment variable")
+		}
+		env["SCW_ACCESS_KEY"] = scwAccessKey
+		scwSecretKey := os.Getenv("SCW_SECRET_KEY")
```
Comment (remyleone) [inline pkg/model/bootstrapscript.go:233]:
```suggestion
		env["SCW_ACCESS_KEY"] = scwAccessKey
		
		scwSecretKey := os.Getenv("SCW_SECRET_KEY")
```

File: pkg/model/bootstrapscript.go [hunk 4]
```diff
@@ -213,6 +214,34 @@ func (b *BootstrapScript) buildEnvironmentVariables(cluster *kops.Cluster) (map[
...
+		if scwSecretKey != "" {
+			return nil, fmt.Errorf("SCW_SECRET_KEY has to be set as an environment variable")
+		}
+		env["SCW_SECRET_KEY"] = scwSecretKey
+		scwProjectID := os.Getenv("SCW_DEFAULT_PROJECT_ID")
```
Comment (remyleone) [inline pkg/model/bootstrapscript.go:238]:
```suggestion
		env["SCW_SECRET_KEY"] = scwSecretKey
		
		scwProjectID := os.Getenv("SCW_DEFAULT_PROJECT_ID")
```

File: nodeup/pkg/bootstrap/install.go [hunk 5]
```diff
@@ -153,6 +153,14 @@ func (i *Installation) buildEnvFile() *nodetasks.File {
 		envVars["AZURE_STORAGE_ACCOUNT"] = os.Getenv("AZURE_STORAGE_ACCOUNT")
 	}
 
+	if os.Getenv("SCW_ACCESS_KEY") != "" {
```
Comment (remyleone) [inline nodeup/pkg/bootstrap/install.go:156]: Should it be secret key? It is always required to have a secret key but not always to have an access key

File: pkg/apis/kops/v1alpha3/cluster.go [hunk 6]
```diff
@@ -238,6 +238,8 @@ type CloudProviderSpec struct {
 	Hetzner *HetznerSpec `json:"hetzner,omitempty"`
 	// Openstack configures the Openstack cloud provider.
 	Openstack *OpenstackSpec `json:"openstack,omitempty"`
+	// Scaleway configures the Scaleway cloud provider.
+	Scaleway *ScalewaySpec `json:"scaleway,omitempty"`
```
Comment (hakman) [inline pkg/apis/kops/v1alpha3/cluster.go:242]: Could you add this to the `v1alpha2/cluster.go` too?

Comment (hakman) [reply pkg/apis/kops/v1alpha3/cluster.go:242]: Please ignore this.

**Review by hakman — APPROVED** [review]
Thanks @Mia-Cross & @remyleone. Looks like a great start. 👍

**Pull request merged**